File: hw/chicaiza/u2/hw19Inheritance/FarmSystemJava/src/main/java/ec/edu/espe/farmsystem/view/FarmSystem.java
package ec.edu.espe.farmsystem.view;
import ec.edu.espe.farmsystem.model.Cage;
import ec.edu.espe.farmsystem.model.Chicken;
import ec.edu.espe.farmsystem.model.FarmAnimal;
import ec.edu.espe.farmsystem.model.Location;
import ec.edu.espe.farmsystem.model.Cow;
import ec.edu.espe.farmsystem.model.Pig;
import ec.edu.espe.farmsystem.model.Sheep;
import java.util.Date;


/**
 *
 * @author Mathews Pastor, Poower Rangers of Programing, @ESPE
 */
public class FarmSystem {
   
    public static void main(String[] args) {
        
        String breed;
        Date date ;
        String gender;
        boolean isAbleToReproduce;
        float weight;
        Cage cage;
        Location location;
        int xCoordinate;
        int yCoordinate;
        
        
        xCoordinate = 10;
        yCoordinate = 10;
        weight =10.4F;
        gender ="male";
        isAbleToReproduce= false;
        
        

       
        
        
        date = new Date(2025 , 2,1);
        breed ="Holstein";
        location = new Location(xCoordinate, yCoordinate);
        cage = new Cage(1, "stable for cows", 2, location);
        
        
        FarmAnimal farmAnimal;
        
        farmAnimal = new Chicken(true, 0, 1, breed, date, gender, isAbleToReproduce, weight, cage);
        
        System.out.println("farmAnimal -->"+farmAnimal);
        
        
        
    
        farmAnimal= new Cow(true, weight, 2, breed, date, gender, isAbleToReproduce, weight, cage);
    
        System.out.println("farmAniaml -->"+farmAnimal);
        
        
        
        date = new Date(2025 , 2,4);
        
        farmAnimal=new Sheep(date, 3, breed, date, gender, isAbleToReproduce, weight, cage);
        
        System.out.println("FarmAnimal -->"+farmAnimal);
        
        
        farmAnimal=new Pig(4, breed, date, gender, isAbleToReproduce, weight, cage);
        
        System.out.println("FarmAnimal -->"+farmAnimal);
        
        
    
    }
    
 
}

File: hw/carvajal/u2/hw19Inheritance/JSFarmSystem/view/farm_system.js
import { Cage } from "../model/Cage.js";
import { Location } from "../model/Location.js";
import { Chicken } from "../model/Chicken.js";
import { Pig } from "../model/Pig.js";
import { Cow } from "../model/Cow.js";
import { Sheep } from "../model/Sheep.js";

function main() {

    let xCoordinate = 10;
    let yCoordinate = 20;
    let weight = 10.4;
    let gender = "male";
    let isAbleToReproduce = false;

    let date = new Date(2025, 1, 1);
    let breed = "Holstein";

    let location = new Location(xCoordinate, yCoordinate);
    let cage = new Cage(1, "STABLE FOR COWS", 2, location);

    let farmAnimal;

    farmAnimal = new Chicken(true, 0, 1, breed, date, gender, isAbleToReproduce, weight, cage);
    console.log("farmAnimal --->", farmAnimal.toString());

    farmAnimal = new Pig(2, breed, date, gender, isAbleToReproduce, weight, cage);
    console.log("farmAnimal --->", farmAnimal.toString());

    farmAnimal = new Cow(true, 5, 3, breed, date, gender, isAbleToReproduce, weight, cage);
    console.log("farmAnimal --->", farmAnimal.toString());

    farmAnimal = new Sheep(date, 5, 4, breed, date, gender, isAbleToReproduce, weight, cage);
    console.log("farmAnimal --->", farmAnimal.toString());
}

main();
